Source/SwiftLintBuiltInRules/Rules/Style/TrailingClosureRule.swift

```diff
@@ -18,127 +19,81 @@ struct TrailingClosureRule: OptInRule {
             Example("offsets.sorted { $0.offset < $1.offset }"),
             Example("foo.something({ return 1 }())"),
             Example("foo.something({ return $0 }(1))"),
-            Example("foo.something(0, { return 1 }())")
+            Example("foo.something(0, { return 1 }())"),
+            Example("for x in list.filter({ $0.isValid }) {}"),
+            Example("if list.allSatisfy({ $0.isValid }) {}"),
+            Example("foo(param1: 1, param2: { _ in true }, param3: 0)"),
+            Example("foo(param1: 1, param2: { _ in true }) { $0 + 1 }"),
+            Example("foo(param1: { _ in false }, param2: { _ in true })"),
+            Example("foo(param1: { _ in false }, param2: { _ in true }, param3: { _ in false })"),
+            Example("""
+            if f({ true }), g({ true }) {
+                print("Hello")
+            }
+            """),
+            Example("""
+            for i in h({ [1,2,3] }) {
+                print(i)
+            }
+            """)
         ],
         triggeringExamples: [
             Example("↓foo.map({ $0 + 1 })"),
             Example("↓foo.reduce(0, combine: { $0 + 1 })"),
             Example("↓offsets.sorted(by: { $0.offset < $1.offset })"),
-            Example("↓foo.something(0, { $0 + 1 })")
+            Example("↓foo.something(0, { $0 + 1 })"),
+            Example("↓foo.something(param1: { _ in true }, param2: 0, param3: { _ in false })"),
+            Example("""
+            for n in list {
+                ↓n.forEach({ print($0) })
+            }
+            """)
```

Comment:
Let's exclude this example from being documented. It isn't special from a user's perspective:
```suggestion
            """, excludeFromDocumentation: true)
```

Reply:
Fixed at e1efa80